### PR TITLE
Development/mesa compositor

### PR DESCRIPTION
**Pull request by bramoosterhuis**

### CODE REVIEW

File: Compositor/lib/Mesa/include/CompositorTypes.h [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        , _modifiers(1, DefaultModifier)
+    {
+    }
+
+    const uint32_t& Type() const
+    {
+        return _fourcc;
+    }
+
+    /*
```
Comment (mhughesacn) [inline Compositor/lib/Mesa/include/CompositorTypes.h:86]: The comment at 86-94 matches widely, but no actual code match.  Please change l.87 to:
Format Modifiers:
(from drm_fourcc.h, Copyright 2011 Intel Corporation (MIT Licensed))

and append to top level NOTICE:
Copyright 2011 Intel Corporation
Licensed under the MIT License

And append an MIT license to LICENSE (using the generic fields  for YEAR and COPYRIGHT HOLDER)

File: Compositor/lib/Mesa/renderer/src/GL/EGL.h [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+        inline static void SaveContext(EglContext& context)
+        {
+            context.display = eglGetCurrentDisplay();
+            context.context = eglGetCurrentContext();
+            context.draw_surface = eglGetCurrentSurface(EGL_DRAW);
+            context.read_surface = eglGetCurrentSurface(EGL_READ);
+        }
+
+        inline static bool RestoreContext(const EglContext& context)
```
Comment (mhughesacn) [inline Compositor/lib/Mesa/renderer/src/GL/EGL.h:100]: Unless you wrote this and wlroots took it from you, then lines 100
-109 are from https://github.com/swaywm/wlroots/tree/master/render and need a credit.  Please add before line 100:

Copyright (c) 2017, 2018 Drew DeVault
Copyright (c) 2014 Jari Vetoniemi
Licensed under the MIT License

And append the same lines to NOTICE.

And append an MIT license to LICENSE (keeping generic fields for YEAR and COPYRIGHT), if not already added from another comment.

File: Compositor/lib/Mesa/backend/src/DRM/DrmMonitor.h [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+
+        void Cards(std::vector<node_t>& _nodes)
+        {
+            ASSERT(_udev != nullptr);
+
+            _nodes.clear();
+
+            struct udev_enumerate* cards = udev_enumerate_new(_udev);
+
+            if (cards != nullptr) {
```
Comment (mhughesacn) [inline Compositor/lib/Mesa/backend/src/DRM/DrmMonitor.h:162]: Appears to be derived from wlroots as in EGL.h comment.  If so, please add a similar credit before line 162:  (If not borrowing from wlroots, then you need to explain the exact match of the comment at line 180.)

Based on code from:
Copyright (c) 2017, 2018 Drew DeVault
Copyright (c) 2014 Jari Vetoniemi
Licensed under the MIT License

NOTICE & LICENSE changes already noted in EGL.h

File: Compositor/lib/Mesa/renderer/src/GL/RenderAPI.h [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+        PFNGLDEBUGMESSAGECALLBACKKHRPROC glDebugMessageCallbackKHR;
+        PFNGLDEBUGMESSAGECONTROLKHRPROC glDebugMessageControlKHR;
+        PFNGLPOPDEBUGGROUPKHRPROC glPopDebugGroupKHR;
+        PFNGLPUSHDEBUGGROUPKHRPROC glPushDebugGroupKHR;
+        PFNGLGETGRAPHICSRESETSTATUSKHRPROC glGetGraphicsResetStatusKHR;
+    }; // class GL
+
+    class EGL {
+    public:
+#define CASE_STR(value) \
```
Comment (mhughesacn) [inline Compositor/lib/Mesa/renderer/src/GL/RenderAPI.h:244]: Are lines 244-270 original?  Protex finds one match.  However, the code is very boilerplate (just a list of EGL errors) and a macro which is a straightforward use of the C++ preprocessor.  So easier to rewrite: Please rename to CASE_TO_STRING or similar and add a comment on line 243 to say you are using the stringification operator.

File: Compositor/lib/Mesa/backup/Mesa-working.cpp [hunk 5]
```diff
@@ -0,0 +1,3454 @@
...
+#define _EGL_SIGNALED EGL_SIGNALED_KHR
+#define _EGL_SYNC_FLUSH_COMMANDS_BIT EGL_SYNC_FLUSH_COMMANDS_BIT_KHR
+
+            using EGLAttrib = EGLint;
+#endif
+            using EGLuint64KHR = khronos_uint64_t;
+
+            EGLContext _ctx = EGL_NO_CONTEXT;
+
+#define CASE_STR(value) \
```
Comment (mhughesacn) [inline Compositor/lib/Mesa/backup/Mesa-working.cpp:539]: Same comments as in RendererAPI.h

**Review by mhughesacn — CHANGES_REQUESTED** [review]
Hi Bram,
Real code matches were detected here - please see my review comments.
Thank you,
Martin Hughes, RDK CMF Compliance Team

File: Compositor/lib/Mesa/include/CompositorTypes.h (hunk 1 above)
Comment (bramoosterhuis) [reply Compositor/lib/Mesa/include/CompositorTypes.h:86]: I removed the comment and point the reader to to the drm_fourcc.h header to find the details.

File: Compositor/lib/Mesa/renderer/src/GL/RenderAPI.h (hunk 4 above)
Comment (bramoosterhuis) [reply Compositor/lib/Mesa/renderer/src/GL/RenderAPI.h:244]: Fixed

File: Compositor/lib/Mesa/backend/src/DRM/DrmMonitor.h (hunk 3 above)
Comment (bramoosterhuis) [reply Compositor/lib/Mesa/backend/src/DRM/DrmMonitor.h:162]: This header is a left over of a test, it's not used therefore I deleted the file.

File: Compositor/lib/Mesa/renderer/src/GL/EGL.h (hunk 2 above)
Comment (bramoosterhuis) [reply Compositor/lib/Mesa/renderer/src/GL/EGL.h]: I introduced a C++ object to save and restore the EGL context, it also works a bit more easier that the C-ish way.

**Review by mhughesacn — APPROVED** [review]
Hi Bram: Thank you for making the changes.  Blackduck is happy!

**Pull request merged**